src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/service/impl/MessageProcessorServiceImpl.java

```diff
@@ -39,6 +41,8 @@
     @Autowired
     private MessageProducer messageProducer;
 
+    private List<String> failedMessageIds = new ArrayList<>();
```

Comment:
Having the array outside of the method I think is going to cause thread safety issues as the listeners we are are using (and kafka consumers) are not thread safe (and this is called by the listener). We'd have to find a way of making this a local variable.

Reply:
Or could use a vector.

Reply:
Local variables make this much harder to test.

Reply:
the vector's methods are synchronised but it will still be the same vector being shared by multiple threads. I'll have a think and see if there's a way we can use local vars.

Reply:
Don't use `Vector` it's obsolete. If you need a thread-safe list use `synchronizedList` method of the `Collections` class. personally I'd be tempted to use follow Mark's advice and use a local variable.

Reply:
Have updated this with a local variable.